shell/shared/grain.js

```diff
@@ -71,6 +73,20 @@ if (Meteor.isServer) {
            ];
   });
 
+  Meteor.publish("tokenInfo", function (token) {
+    // Allows the client side to map a raw token to its entry in ApiTokens.
+
+    var hashedToken = Crypto.createHash("sha256").update(token).digest("base64");
+    var apiToken = ApiTokens.findOne({_id: hashedToken});
```

Comment:
We need to restrict what fields are revealed here. For example, `objectId` may contain secrets private to a grain which the token bearer is not intended to be able to see.